posts/brugner/1.sqlserver-docker-m1.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Running SQL Server on Docker on a M1 Mac
+published: false
+description: Microsoft hasn't released an ARM-based SQL Server Docker image, what can we do on a M1 Mac if we want to run the popular RDBMS?
+tags: "sqlserver, docker, m1, arm"
+cover_image: ./assets/cover1.png
+---
+
+In 2020 Apple decided to replace the Intel chipsets with one of their own making, that has been leading performance benchmarks ever since. But these chips are based on the **ARM architecture** instead of the widespread x64 used by its predecesor so, if we want to run software natively and take advantage of all of its power, programs should be ARM ready. This also goes to **Docker** images.
+
+Unfortunately, as of 2022, Microsoft hasn't released yet a fully fledged SQL Server Docker image for ARM processors. What can we do about it?
+
+## Enter Azure SQL Edge
+
+The alternative is [Azure SQL Edge](https://azure.microsoft.com/en-us/products/azure-sql/edge/), a lighter version of the popular database engine based on the ARM architecture. Even though this version is limited in order to run better on IoT devices, it comes with some additional features like built-in machine learning and data streaming capabilities. Let's see how to run a container with it.
+
+## Install Docker
+
+There are many ways to install Docker. I prefer downloading Docker for Desktop from their [official site](https://www.docker.com/products/docker-desktop/). Remember to choose the **Apple Chip** version.
+
+![Download Docker for Desktop](./assets/download-docker-desktop.png)
+
+After the installation is done, open Docker For Desktop. You'll be welcomed by a home page that shows no containers running.
+
+![Docker For Desktop](./assets/docker-for-desktop-installed.pngg)
+
+## Running Azure SQL Edge on a container
+
+Open your favorite terminal and run the following commands.
+
+**Pull the image from the registry**
+`docker pull mcr.microsoft.com/azure-sql-edge`
+
+**Run the container**
+`docker run --cap-add SYS_PTRACE -e 'ACCEPT_EULA=1' -e 'MSSQL_SA_PASSWORD=Str#ng_Passw#rd' -p 57000:1433 --name azuresqledge -d mcr.microsoft.com/azure-sql-edge`
+
+Important parameters to notice here:
+
+- **ACCEPT_EULA**: accepts the license agreement.
+- **MSSQL_SA_PASSWORD**: sets the password for the sa user.
+- **p**: maps your machine's port to a container's port, in this case 1433 where SQL runs.
+- name: any name you want to use to identify this container.
+
+After running these commands successfully, go to Docker For Desktop again and you'll see the newly created container up and running.
+
+![Container running](./assets/container-running.png)
+
+## Connecting to the server
+
+In order to connect to database server, download [Azure Data Studio](https://learn.microsoft.com/en-us/sql/azure-data-studio/download-azure-data-studio?view=sql-server-ver16) **arm64** version.
```

Comment:
Remove the `i18n` from MSFT URLs, their resolve to the correct language for the reader
```suggestion
In order to connect to database server, download [Azure Data Studio](https://learn.microsoft.com/sql/azure-data-studio/download-azure-data-studio#download-azure-data-studio) **Apple Silicon** version.
```
`arm64` is not present in this page, it uses `Apple Silicon`